File: smartphone.cpp
#include<stdio.h>
#include<stdlib.h>
#include<string.h>

const int lim = 10;
struct smart
{
 char modl[30], cmp[30];
 float dig;
 int dat;
};
smart cat[lim];
int i,n = 0;

void add()
{
 system("clear");
 if (n==(lim-1))
 printf("Каталог заполнен\n");
 else
 {
  n++;
  printf("Введите производителя: ");
  scanf("%s", cat[n].cmp);
  printf("\nВведите модель: ");
  scanf("%s", cat[n].modl);
  printf("\nВведите диагональ: ");
  scanf("%f", &cat[n].dig);
 }
}

void odin(int a)
{
 printf("Производитель: %s\n", cat[n].cmp);
 printf("Модель: %s\n", cat[n].modl);
 printf("Диагональ: %f\n", cat[n].dig);
}
void allin()
{
 system("clear");
 if (n==0)
 printf("Пусто");
 else
 for (i=0;i<n;i++)
 odin(i);
}
void find()
{
 char b[30];
 int c;
 system("clear");
 printf("Введите модель телефона: ");
 c = 0;
 scanf("%s", b);
 for(i=0;i<n;i++)
 {
  if (!strcmp(b, cat[i].cmp))
  {
   c=1;
   odin(i);
  }
  if (c==0)
  printf("Не найдено");
 }
}

int menu()
{
 int d;
 system("clear");
 printf("1 - добавить\n");
 printf("2- показать всё\n");
 printf("3 - показать по производителю\n");
 printf("4 - выход\n");
 scanf("%d", &d);
 return d;
}

int main()
{
 n = 0;
 while (1)
 {
  switch(menu())
  {
   case 1:
          add();
          break;
   case 2:
          allin();
          printf("Нажмите Ctrl+C...\n");
          system("pause");
          break;
   case 3:
          find();
          printf("Нажмите Ctrl+C...\n");
          system("pause");
          break;
   case 4:
          return 0;
  }
 }
 return 0;
}

File: 1kurs/guide/smartphone.pas
uses crt;
const lim = 10;


type smart = record
    cmp,modl: string[30];
    dig: single;
    bat: integer;
    end;


var cat: array[1..lim] of smart;
    i,n: integer;

Procedure add;
begin
     clrscr;
     if n=lim then
       writeln('База переполнена.')
       else
       begin
         n:=n+1;
       write('Введите производителя: ');
       readln(cat[n].cmp);
       write('Введите модель: ');
       readln(cat[n].modl);
       write('Введите диагональ: ');
       readln(cat[n].dig);
       write('Введите ёмкость батареи: ');
       readln(cat[n].bat);
       end;
end;


Procedure odin (a:integer);
begin
     writeln('Производитель: ',cat[a].cmp);
     writeln('Модель: ',cat[a].modl);
     writeln('Диагональ: ',cat[a].dig:0:2);
     writeln('Ёмкость: ',cat[a].bat);
     writeln;
end;

Procedure allin;
begin
     clrscr;
     If n=0 then
       writeln('пусто')
     else
          for i:=1 to n do
            odin(i);
end;


Procedure find;
var b: string;
    c: boolean;
begin
     clrscr;
     writeln('Поиск производителя:');
     c:=false;
     readln(b);
     for i:=1 to n do
     if b=cat[i].cmp then
     begin
          c:=true;
          odin(i);
     end;
     if c = false then
       writeln('Такой производитель не найден.');
end;

    
function menu: integer;
var
   d:integer;
begin
     clrscr;
     writeln('Приветствуем Вас в нашем магазине смартфнов!');
     writeln(' 1- добавить смартфон в каталог');
     writeln(' 2- показать весь каталог');
     writeln(' 3 - поиск по производителю');
     writeln(' 4 - выход');
     writeln('Выберете действие');
     readln(d);
     menu:=d;
end;

begin
     n:=0;
     while true do 
     begin
          case menu of
            1:add;
            2:begin allin;
          writeln('Нажмите клавишу');
          readkey;
          end;
          3:begin
          find;
          writeln('нажмите клавишу');
          readkey;
          end;
          4:break;
          end;
    end;
end.
          